### PR TITLE
[Merged by Bors] - feat: port NumberTheory.Padics.PadicVal

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/Padics/PadicVal.lean [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+
+The `p`-adic valuation on `ℚ` is the difference of the multiplicities of `p` in the numerator and
+denominator of `q`. This function obeys the standard properties of a valuation, with the appropriate
+assumptions on `p`. The `p`-adic valuations on `ℕ` and `ℤ` agree with that on `ℚ`.
+
+The valuation induces a norm on `ℚ`. This norm is defined in padic_norm.lean.
+
+## Notations
+
+This file uses the local notation `/.` for `rat.mk`.
```
Comment (RemyDegenne) [inline Mathlib/NumberTheory/Padics/PadicVal.lean:27]:
```suggestion
This file uses the local notation `/.` for `Rat.mk`.
```

File: Mathlib/NumberTheory/Padics/PadicVal.lean [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+open multiplicity
+
+variable {p : ℕ} [hp : Fact p.Prime]
+
+/-- The multiplicity of `p : ℕ` in `a : ℤ` is finite exactly when `a ≠ 0`. -/
+theorem finite_int_prime_iff {a : ℤ} : Finite (p : ℤ) a ↔ a ≠ 0 := by
+  simp [finite_int_iff, hp.1.ne_one]
+#align padic_val_rat.finite_int_prime_iff padicValRat.finite_int_prime_iff
+
+/-- A rewrite lemma for `padic_val_rat p q` when `q` is expressed in terms of `rat.mk`. -/
```
Comment (RemyDegenne) [inline Mathlib/NumberTheory/Padics/PadicVal.lean:264]:
```suggestion
/-- A rewrite lemma for `padic_val_rat p q` when `q` is expressed in terms of `Rat.mk`. -/
```

File: Mathlib/NumberTheory/Padics/PadicVal.lean [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+    multiplicity.mul' (Nat.prime_iff_prime_int.1 hp.1)]
+  rw [Nat.cast_add, Nat.cast_add]
+  simp_rw [Int.coe_nat_multiplicity p q.den]
+  ring
+  -- Porting note: was
+  -- simp only [hc1, hc2, multiplicity.mul' (Nat.prime_iff_prime_int.1 hp.1),
+  --   hp.1.ne_one, hqz, pos_iff_ne_zero, Int.coe_nat_multiplicity p q.den
+#align padic_val_rat.defn padicValRat.defn
+
+/-- A rewrite lemma for `padic_val_rat p (q * r)` with conditions `q ≠ 0`, `r ≠ 0`. -/
```
Comment (RemyDegenne) [inline Mathlib/NumberTheory/Padics/PadicVal.lean:286]:
```suggestion
/-- A rewrite lemma for `padicValRat p (q * r)` with conditions `q ≠ 0`, `r ≠ 0`. -/
```
Could you fix that name and others everywhere in the file?

Comment (Parcly-Taxel) [reply Mathlib/NumberTheory/Padics/PadicVal.lean]: All done.

**Comment by RemyDegenne** [comment]
Thanks!
bors r+

**Pull request closed without merging**